app/src/main/assets/layouts/numpad.xml

```diff
@@ -1,15 +1,23 @@
 <?xml version="1.0" encoding="utf-8" ?>
 <vertical>
     <horizontal>
-        <button text="7" /> <button text="8" /> <button text="9" />
+        <button text="7" />
+        <button text="8" />
+        <button text="9" />
     </horizontal>
     <horizontal>
-        <button text="4" /> <button text="5" /> <button text="6" />
+        <button text="4" />
+        <button text="5" />
+        <button text="6" />
     </horizontal>
     <horizontal>
-        <button text="1" /> <button text="2" /> <button text="3" />
+        <button text="1" />
+        <button text="2" topRightText="sin" topRightAction="sin(@)"/>
+        <button text="3" topRightText="π"/>
```

Comment:
This looks like one needs to swipe diagonally. Feels a bit weird, to be honest. I would rather add top/bottom/left/right for swiping and something like "secondary" for long-press. That one can be displayed in the top right but I wouldn't explicitly call it "top right" in the layout because RTL languages might want to have it in the other corner

Reply:
Long press is already implemented via the key `longPressAction`.

To me, the diagonal actions feel quite natural. Additionally, having the labels at the corners allows for tighter layouts (e.g. trying to fit a `cos` to the left and a `tan` to the right of a button is likely pretty crowded, but I guess this scenario could work with labels at the corners)

However, we can certainly try out a top/bottom/left/right layout. Maybe, let me first fix the styling of the new calculator buttons. After that we can quickly change the `calculator_button.xml` to see which one feels better?

Reply:
> Long press is already implemented via the key longPressAction.

Where is the long press text displayed? Would that one be at the top, then?

Reply:
Ahh, sorry, I did not comment on that part. Currently the long press is only used for the "clear all" action when long pressing the backspace button (like you implemented in one of your commits. Thanks for that :+1: )
I thought that the long press opens some kind of popup menu, that lets you choose further actions that are somewhat related to that button (e.g. a long press on the sin button could give access to hyperbolic and inverse sine functions). In that case I think we wouldn't need a separate label? Or maybe we could display a simple caret symbol at the bottom?

Reply:
> like you implemented in one of your commits. 

Now that it is gone (for now), I really miss that feature :D

> I thought that the long press could open some kind of popup menu

Are these alternative functions then described in xml as well? I would vote to do so (then the layouts are fully flexible). Something like `<button longPressAction1="cos(@)" longPressText1="cos" longPressAction2="..."` could be a bit too verbose, though.

Reply:
> > 
> 
> Now that it is gone (for now), I really miss that feature :D

Same for me :D I immediately got used to it!

> Are these alternative functions then described in xml as well?

I would also vote to do so. I thought about realizing that using child elements within the button element? That would also provide the flexibility for divider elements or similar within the popup menu.
In that case, the `longPressAction` is of cause not needed anymore. How do you think about this?

Reply:
Hmm, so would it show the menu only when there are 2 or more alternative actions? Otherwise, long pressing the delete button would open a list instead of clearing directly.